src/Parse/Date.php

```diff
@@ -679,54 +679,66 @@ public function add_callback($callback)
+                                :?
+                                (?P<second>[0-9]{2})
+                                (?:
+                                    .
+                                    (?P<second_fraction>[0-9]*)
+                                )?
+                            )?
+                        )?
+                        (?:
+                            (?P<zulu>Z)
+                            |   (?P<tz_sign>[+\-])
+                                (?P<tz_hour>[0-9]{1,2})
+                                :?
+                                (?P<tz_minute>[0-9]{1,2})
+                        )
+                    )?
+                )?
+            )?
+            $
+            /x
+PCRE;
         if (preg_match($pcre, $date, $match)) {
-            /*
-            Capturing subpatterns:
-            1: Year
-            2: Month
-            3: Day
-            4: Hour
-            5: Minute
-            6: Second
-            7: Decimal fraction of a second
-            8: Zulu
-            9: Timezone ±
-            10: Timezone hours
-            11: Timezone minutes
-            */
-
-            // Fill in empty matches
-            for ($i = count($match); $i <= 3; $i++) {
-                $match[$i] = '1';
-            }
-
-            for ($i = count($match); $i <= 7; $i++) {
-                $match[$i] = '0';
-            }
+            // Fill in empty matches and convert to proper types.
+            $year = (int) $match['year'];
+            $month = isset($match['month']) ? (int) $match['month'] : 1;
+            $day = isset($match['day']) ? (int) $match['day'] : 1;
+            $hour = isset($match['hour']) ? (int) $match['hour'] : 0;
+            $minute = isset($match['minute']) ? (int) $match['minute'] : 0;
+            $second = isset($match['second']) ? (int) $match['second'] : 0;
+            $second_fraction = isset($match['second_fraction']) ? ((int) $match['second_fraction']) / (10 ** strlen($match['second_fraction'])) : 0;
+            $tz_sign = ($match['tz_sign'] ?? '') === '-' ? -1 : 1;
+            $tz_hour = isset($match['tz_hour']) ? (int) $match['tz_hour'] : 0;
+            $tz_minute = isset($match['tz_minute']) ? (int) $match['tz_minute'] : 0;
 
             // Numeric timezone
-            if (isset($match[9]) && $match[9] !== '') {
-                $timezone = $match[10] * 3600;
-                $timezone += $match[11] * 60;
-                if ($match[9] === '-') {
-                    $timezone = 0 - $timezone;
-                }
-            } else {
-                $timezone = 0;
-            }
+            $timezone = $tz_hour * 3600;
+            $timezone += $tz_minute * 60;
+            $timezone *= $tz_sign;
 
             // Convert the number of seconds to an integer, taking decimals into account
-            $second = round((int)$match[6] + (int)$match[7] / (10 ** strlen($match[7])));
+            $second = round($second + $second_fraction);
```

Comment:
We should cast the returned `float` from `round()` into `int`.

```suggestion
            $second = (int) round($second + $second_fraction);
```

Reply:
Fixed, thanks.